### PR TITLE
WIP - Fix determining app version for chart to pull from test catalog for integration tests

### PR DESCRIPTION
**Pull request by uvegla**
I am not sure this will work, but it bugs me that the integration tests are semi randomly breaking and I think this is the issue.

## Checklist

- [ ] Update changelog in CHANGELOG.md.

### CODE REVIEW

File: integration/key/key.go [hunk 1]
```diff
@@ -5,6 +5,9 @@ package key
 
 import (
 	"fmt"
+	"strings"
+
+	"github.com/giantswarm/app-operator/v6/integration/env"
```
Comment (MarcelMue) [inline integration/key/key.go:10]: CI is unhappy about your import order.

**Review by MarcelMue — APPROVED** [review]
Generally makes sense to me.

File: integration/key/key.go (hunk 1 above)
Comment (uvegla) [reply integration/key/key.go:10]: Yea, fixed it,  thanks!

**Comment by uvegla** [comment]
Even tho it passes this is not okay. 😢  I realised that the `-dev` postfixed version is added later by some automation so integration tests ran against `6.0.0` and not this commit. @MarcelMue Do you know of some convenient way to figure if it is a release build or not?

**Comment by MarcelMue** [comment]
> Even tho it passes this is not okay. cry I realised that the `-dev` postfixed version is added later by some automation so integration tests ran against `6.0.0` and not this commit. @MarcelMue Do you know of some convenient way to figure if it is a release build or not?

Is it maybe because this PR isn't merged yet?
https://github.com/giantswarm/app-operator/pull/1011

**Comment by uvegla** [comment]
> > Even tho it passes this is not okay. cry I realised that the `-dev` postfixed version is added later by some automation so integration tests ran against `6.0.0` and not this commit. @MarcelMue Do you know of some convenient way to figure if it is a release build or not?
> 
> Is it maybe because this PR isn't merged yet? #1011

Hah, nice, merged. I am still not super happy tho, as it is easy to miss or not merge those at all even and I can merge code that has falsely passing integration tests. 🤔

**Comment by MarcelMue** [comment]
> > > Even tho it passes this is not okay. cry I realised that the `-dev` postfixed version is added later by some automation so integration tests ran against `6.0.0` and not this commit. @MarcelMue Do you know of some convenient way to figure if it is a release build or not?
> > 
> > 
> > Is it maybe because this PR isn't merged yet? #1011
> 
> Hah, nice, merged. I am still not super happy tho, as it is easy to miss or not merge those at all even and I can merge code that has falsely passing integration tests. thinking

I do not disagree - I just don't have a good solution in my brain :D

**Comment by uvegla** [comment]
There is no way to differentiate on a master branch build if it is an actual release or building unreleased changes, see:
- actual release merge (failed, integration test tries to pull by SHA that is not uploaded to the test catalog): https://app.circleci.com/pipelines/github/giantswarm/app-operator/3952/workflows/6cc0974b-faf6-4057-908b-9d03eee9d8a3/jobs/20723/parallel-runs/0/steps/0-99
- unreleased change merge (success, the artefact was uploaded to test catalog via SHA): https://app.circleci.com/pipelines/github/giantswarm/app-operator/3961/workflows/46212ec7-d684-4170-8286-ea3650d55e4d/jobs/20785/parallel-runs/0/steps/0-99

**Comment by uvegla** [comment]
One drawback this brings is that locally now you also have to set `CIRCLE_BRANCH` not just `CIRCLE_SHA1`. On CircleCI these are always set.

**Comment by MarcelMue** [comment]
> One drawback this brings is that locally now you also have to set `CIRCLE_BRANCH` not just `CIRCLE_SHA1`. On CircleCI these are always set.

I don't think this is a big drawback. It generally makes sense to me.

**Review by MarcelMue — APPROVED** [review]
I don't see an issue with this approach. It's a definite improvement to me

**Pull request merged**